src/dodal/beamlines/i22.py

```diff
@@ -9,6 +14,14 @@
 set_utils_beamline(BL)
 
 
+@lru_cache
+def get_directory_provider() -> DirectoryProvider:
+    return StaticVisitDirectoryProvider(
+        BL,
+        "/data/i22/2024/VISIT",
```

Comment:
I think I must not be understanding this PR at all, so bear with me, but:
- Why are we hard-coding `2024`?
- What is `VISIT`? Is that a placeholder that's means to get magically replaced elsewhere?
- If we're `lru_caching` this doesn't that mean `blueapi` needs a full restart just to change visits? That seems suboptimal.
- In p38 we're *also* literally hard-coding `cm37282-2`, that really doesn't seem right??

Reply:
You're understanding perfectly. 

Unfortunately right now we can't authenticate the blueapi user, so we cannot (according to the security policy) expose the list of visits that the user could write to (because the visits that user X is on is privileged information that user Y is not allowed to access), so we set blueapi up for a single visit. This is why the year, visit are hardcoded right now. VISIT is just a placeholder for comment and review at the moment. The p38/p45 commissioning visits are the correct ones for writing data into currently.

This is how the hardcoding was done in November (in one part, beamline and visit). Eventually the beamline will be hardcoded and the visit will be updating with (potentially) every request. This is a step to get there. 

The lru_caching will still be on the DirectoryProvider, but the DirectoryProvider will return a new DirectoryInfo that has relevant information for the current visit for each Run.

Reply:
An option if we don't want to hard-code it here is to use an environment variable or similar if anyone has thoughts on that

Reply:
I was originally imagining you'd have a plan exposed by blueAPI that could set the year/visit - because then it can be changed at runtime and avoids needing to edit `dodal` code just to change experiments.

But tbh I'm happy with just documentation if this is an "intermediate" solution.